src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -436,6 +531,176 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 }
 
 
+/*
+ * CreateCitusLocalTable is the internal method to create citus local table's
+ * shard and metadata.
+ * Note that this function does not perform any validation on the table as it
+ * is already done in create_citus_local_table.
+ */
+void
+CreateCitusLocalTable(Oid relationOid)
+{
+	/*
+	 * Make sure that existing reference tables have been replicated to all
+	 * the nodes such that we can create foreign keys and joins work immediately
+	 * after creation.
+	 */
+	EnsureReferenceTablesExistOnAllNodes();
```

Comment:
Could we ever be in a state where reference  tables are not replicated?

Reply:
As https://github.com/citusdata/citus/pull/3637 claims, yes